From d08d1c0a7c5601c0bd81067276af4e04800b46a9 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Fri, 19 Mar 2021 12:23:25 +0530
Subject: [PATCH 736/767] Revert "octeontx2: Re-enable FLR and MBOX interrupts"

commit e46533f7e5359ff1cf1832658aa91666a5ceb80f from
git@git.assembla.com:cavium/WindRiver.linux.git

This reverts commit 748940c500679b123cda545d9fa118ecca7ad66c.
PF_RST on a FLR is causing RVU_PF_VF_BAR4_ADDR also getting cleared.
Due to that RVU_PF_VF_BAR4_ADDR is also getting cleared which
would break PF::VF communication

Change-Id: I6a5362ec25fa8ea9c3b1a6daa22309ed2330e79d
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/48089
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/soc/marvell/octeontx2-rm/otx2_rm.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-rm/otx2_rm.c b/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
index 8f8067cf4757..ea76bd3db6e4 100644
--- a/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
+++ b/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
@@ -458,15 +458,9 @@ static void rm_pfvf_flr_handler(struct work_struct *work)
 	rm_write64(rm, BLKADDR_RVUM, 0, RVU_PF_VFTRPENDX(idx),
 		   BIT_ULL(vf->intr_idx));
 
-	/* Re-enable MBOX, FLR, and ME interrupt as it gets cleared
-	 * in HWVF_RST reset
-	 */
-	rm_write64(rm, BLKADDR_RVUM, 0, RVU_PF_VFFLR_INT_ENA_W1SX(idx),
-		   BIT_ULL(vf->intr_idx));
+	/* Re-enable ME interrupt as it gets cleared in HWVF_RST reset */
 	rm_write64(rm, BLKADDR_RVUM, 0, RVU_PF_VFME_INT_ENA_W1SX(idx),
 		   BIT_ULL(vf->intr_idx));
-	rm_write64(rm, BLKADDR_RVUM, 0, RVU_PF_VFPF_MBOX_INT_ENA_W1SX(idx),
-		   BIT_ULL(vf->intr_idx));
 }
 
 static void rm_pfvf_mbox_handler_up(struct work_struct *work)
-- 
2.31.1

